### PR TITLE
Added @lambda-lambda-lambda/cli types

### PR DESCRIPTION
**Pull request by nuxy**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:

- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/lambda-lambda-lambda__cli/lambda-lambda-lambda__cli-tests.ts
1. types/lambda-lambda-lambda__cli/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against ff4dd4c23389f97fcc4690fb47dd68ae1b4bb4fc
</p>

**Comment by peterblazejewicz** [comment]
hey, the module repo contains already TS files, what's the status?
https://github.com/lambda-lambda-lambda/cli/blob/master/index.d.ts

> [x] The package does not already provide its own types, or cannot have its .d.ts files generated via --declaration

**Comment by nuxy** [comment]
> hey, the module repo contains already TS files, what's the status?

The types in question are to be removed post PR approval/release.

**Comment by peterblazejewicz** [comment]
I believe TS/TD team encourages everyone to stick to handbook (your code contains .d.ts? ship it with our code):
https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html#including-declarations-in-your-npm-package
as this is what people in community expect in first place. Why do you want to have those files here?
(sorry, your PR manifest didn't mention anything on those topics)

File: types/lambda-lambda-lambda__cli/.eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+    "rules": {
+        "@typescript-eslint/no-invalid-void-type": "off"
```
Comment (jakebailey) [inline types/lambda-lambda-lambda__cli/.eslintrc.json:3]: Why is this disabled?

**Comment by nuxy** [comment]
> Why do you want to have those files here?

The motivation behind this was to expose CLI types to third-party integrated scripts (non NPM) without having to install the main package since it comes with a lot of baggage (e.g. templates, images, etc).

That said, after revaluating my options I'm going to self-host the types in it's own repo.

**Pull request closed without merging**